ts/screens/profile/PrivacyMainScreen.tsx

```diff
@@ -219,12 +238,18 @@ const PrivacyMainScreen = ({ navigation }: Props) => {
         value: I18n.t("profile.main.privacy.exportData.title"),
         description: I18n.t("profile.main.privacy.exportData.description"),
         onPress: () => {
-          setRequestProcess(true);
-          dispatch(
-            loadUserDataProcessing.request(
-              UserDataProcessingChoiceEnum.DOWNLOAD
-            )
-          );
+          if (pot.isError(userDataProcessing.DOWNLOAD)) {
+            // eslint-disable-next-line functional/immutable-data
+            canShowTooltipRef.current = true;
+            setRequestProcess(true);
+            dispatch(
+              loadUserDataProcessing.request(
+                UserDataProcessingChoiceEnum.DOWNLOAD
+              )
+            );
+          } else {
+            handleUserDataRequestAlert(UserDataProcessingChoiceEnum.DOWNLOAD);
+          }
```

Comment:
What about wrapping this login inside a function having as parameter the `UserDataProcessingChoiceEnum`. What do you think?

Reply:
Thank you for the hint!

Reply:
Done!